pkg/icingaredis/heartbeat.go

```diff
@@ -175,9 +163,32 @@ func (h *Heartbeat) setError(err error) {
 	h.err = errors.Wrap(err, "heartbeat failed")
 }
 
-func (h *Heartbeat) setMessage(m *v1.StatsMessage) {
-	h.messageMu.Lock()
-	defer h.messageMu.Unlock()
+func (h *Heartbeat) sendEvent(m *HeartbeatMessage) {
+	// Remove any not yet delivered event
+	select {
+	case old := <-h.events:
+		if old != nil {
+			h.logger.Debugw("Previous heartbeat not read from channel",
+				zap.Time("previous", old.received),
+				zap.Time("current", m.received))
+		} else {
+			h.logger.Debug("Previous heartbeat loss event not read from channel")
+		}
+	default:
+	}
+
+	h.events <- m
+}
+
+type HeartbeatMessage struct {
```

Comment:
I don't know if you would have added it later anyway, but the documentation for this struct and its methods is missing.